ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -88,32 +93,158 @@ tests = testGroup "OnDisk" [
+  -- ledger tables and the block application semantics satisfy:
+  --
+  -- * a key is deleted at most once
+  -- * a key is inserted at most once
+  --
+  applyPayload st Tx{consumed, produced} =
+      delete consumed st >>= uncurry insert produced
+    where
+      insert :: Token -> TValue -> UTxTok -> Either TxErr UTxTok
+      insert tok val UTxTok{utxtok, utxhist} =
+        if tok `Set.member` utxhist
+        then Left  $ TokenWasAlreadyCreated tok
+        else Right $ UTxTok { utxtok  = Map.insert tok val utxtok
+                            , utxhist = Set.insert tok utxhist
+                            }
+
+      delete :: Token -> UTxTok -> Either TxErr UTxTok
+      delete tok st'@UTxTok{utxtok} =
+        if tok `Map.member` utxtok
+        then Right $ st' { utxtok = Map.delete tok utxtok }
+        else Left  $ TokenDoesNotExist tok
+
+{-------------------------------------------------------------------------------
+  TestBlock generation
+
+  When we added support for storing parts of the ledger state on disk we needed
+  to exercise this new functionality. Therefore, we modified this test so that
+  the ledger state associated to the test block contained tables (key-value
+  maps) to be stored on disk. This ledger state needs to follow an evolution
+  pattern similar to the UTxO one (see the 'PayloadSemantics' instance for more
+  details). As a result, block application might fail on a given payload.
+
+  The tests in this module assume that no invalid blocks are generated. Thus we
+  have to satisfy this assumption in the block generators. To keep the
+  generators simple, eg independent on the ledger state, we follow this strategy
+  to block generation:
+
+  - The block payload consist of a single transaction:
+      - input: Point
+      - output: (Point, SlotNo)
+  - The ledger state is a map from Point to SlotNo.
+  - We start always in an initial state in which 'GenesisPoint' maps to slot 0.
+  - When we generate a block for point p, the payload of the block will be:
+      - input: point p
+      - ouptput: (point p + 1, slot of point p + 1)
+
+
+  A consequence of adopting the strategy above is that the initial state is
+  coupled to the generator's semantics.
+ -------------------------------------------------------------------------------}
+
+initialTestLedgerState :: UTxTok
+initialTestLedgerState = UTxTok {
+    utxtok = Map.singleton initialToken (pointTValue initialToken)
+  , utxhist = Set.singleton initialToken
+  }
+  where
+    initialToken = Token GenesisPoint
+
+-- | Get the token value associated to a given token. This is coupled to the
+-- generators semantics.
+pointTValue :: Token -> TValue
+pointTValue (Token GenesisPoint)        = TValue 0
+pointTValue (Token (BlockPoint slot _)) = TValue slot
+
 genBlocks ::
      ExtLedgerCfg TestBlock
   -> Word64
-  -> ExtLedgerState TestBlock
+  -> Point TestBlock
   -> [TestBlock]
 genBlocks _   0 _ = []
-genBlocks cfg n l = b:bs
+genBlocks cfg n pt =
```

Comment:
I think this is equivalent and simpler (and can be readily adapted if we revert to carrying the ledgerstate instead of carrying just the point):

```
genBlocks n pt0 =
    take n (go pt0)
  where
    go pt = let b = genBlock pt in b : go (blockPoint pt)
```